File: fatoora-core/examples/validation.rs
// --8<-- [start:example]
pub fn main() {
    use fatoora_core::config::{Config, EnvironmentType};
    use fatoora_core::invoice::validation::validate_xml_invoice_from_str;

    // Doc example: validate XML against the bundled schema.
    let invoice_xml_path = std::path::Path::new(env!("CARGO_MANIFEST_DIR"))
        .join("tests/fixtures/invoices/sample-simplified-invoice.xml");
    let config = Config::new(EnvironmentType::NonProduction);
    let xml = std::fs::read_to_string(&invoice_xml_path).expect("read invoice xml");
    validate_xml_invoice_from_str(&xml, &config).expect("validate invoice");
}
// --8<-- [end:example]

File: bindings/python/examples/validation.py
from pathlib import Path

invoice_xml_path = (
    Path(__file__).resolve().parents[3]
    / "fatoora-core"
    / "tests"
    / "fixtures"
    / "invoices"
    / "sample-simplified-invoice.xml"
)

# --8<-- [start:example]
from fatoora.config import Config
from fatoora.validation import validate_xml_invoice_from_str

# invoice_xml_path = "path/to/invoice.xml"
config = Config()
xml = invoice_xml_path.read_text(encoding="utf-8")

result = validate_xml_invoice_from_str(config, xml)
assert result is True
# --8<-- [end:example]
